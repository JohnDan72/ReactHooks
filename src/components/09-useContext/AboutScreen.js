import React from 'react';
import PropTypes from 'prop-types';


const propTypes = {};
const defaultProps = {};


const AboutScreen = () => {
    return(
        <>
            <h3></h3>
        </>
    );
}

AboutScreen.propTypes = propTypes;
AboutScreen.defaultProps = defaultProps;

export default AboutScreen;