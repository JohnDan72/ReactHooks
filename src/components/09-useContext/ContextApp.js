import React from 'react';
import { Outlet } from 'react-router-dom';
import {  Container, Content, Header } from 'rsuite';
import NavBarContext from '../NavBarContext';
// import PropTypes from 'prop-types';

const propTypes = {};
const defaultProps = {};

const ContextApp = ({ setTheme }) => {


    return (
        <Container>
            <Header >
                <NavBarContext setTheme={setTheme} />
            </Header>
            <Content className="mt-3 container">
                <Outlet />
            </Content>
        </Container>
    );
}

ContextApp.propTypes = propTypes;
ContextApp.defaultProps = defaultProps;

export default ContextApp;