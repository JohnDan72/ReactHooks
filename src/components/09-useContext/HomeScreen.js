import React, { useContext } from 'react';
// import PropTypes from 'prop-types';
import { Divider } from 'rsuite';
import { UserContext } from "./UserContext";
import JSONTree from 'react-json-tree';


const HomeScreen = () => {

    const { user , setUser} = useContext(UserContext);

    console.log(setUser);
    return (
        <>
            <h3>Home page</h3>
            <Divider></Divider>
            <JSONTree data={user}/>
        </>
    );
}


export default HomeScreen;