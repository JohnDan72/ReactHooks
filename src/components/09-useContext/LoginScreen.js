import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {};
const defaultProps = {};

const LoginScreen = () => {
    return <div></div>;
}

LoginScreen.propTypes = propTypes;
LoginScreen.defaultProps = defaultProps;

export default LoginScreen;