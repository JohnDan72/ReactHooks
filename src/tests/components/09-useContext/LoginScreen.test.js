import { mount } from "enzyme";
import { UserContext } from "../../../components/09-useContext/UserContext";
import LoginScreen from "../../../components/09-useContext/LoginScreen";

describe('Pruebas Login Screen', () => {

    const user = {
        name: 'Juan Dan',
        email: 'jgarcialopez837@gmail.com'
    }
    const setUser = jest.fn();

    let wrapper = mount(
        <UserContext.Provider value={{
            user , setUser
        }}>
            <LoginScreen />

        </UserContext.Provider>
    );

    test('debe renderizar correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });


    test('debe ejectuar setUser con argumentos especÃ­ficos', () => {
        wrapper.find('Button').prop('onClick')();

        expect(setUser).toHaveBeenCalledTimes(1);
        expect(setUser).toHaveBeenLastCalledWith({
            id: 123,
            name: 'Juan Daniel Garcia',
            edad: 24
        });
    })
    
    
})
