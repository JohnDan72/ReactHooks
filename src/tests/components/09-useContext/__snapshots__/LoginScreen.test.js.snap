// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Login Screen debe renderizar correctamente 1`] = `
Array [
  <h3>
    Login page
  </h3>,
  <div
    aria-orientation="horizontal"
    className="rs-divider rs-divider-horizontal"
    role="separator"
  />,
  <button
    className="rs-btn rs-btn-primary"
    onClick={[Function]}
    type="button"
  >
    Login
    <span
      className="rs-ripple-pond"
    >
      <span
        className="rs-ripple"
      />
    </span>
  </button>,
  <div
    aria-orientation="horizontal"
    className="rs-divider rs-divider-horizontal"
    role="separator"
  />,
  <ul
    style={
      Object {
        "MozUserSelect": "none",
        "WebkitUserSelect": "none",
        "backgroundColor": "#002b36",
        "border": 0,
        "listStyle": "none",
        "marginBottom": "0.5em",
        "marginLeft": "0.125em",
        "marginRight": 0,
        "marginTop": "0.5em",
        "padding": 0,
      }
    }
  >
    <li
      style={
        Object {
          "marginLeft": 0,
          "paddingLeft": 0,
          "paddingTop": "0.25em",
          "position": "relative",
        }
      }
    >
      <div
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "inline-block",
            "paddingLeft": 0,
            "paddingRight": "0.5em",
          }
        }
      >
        <div
          style={
            Object {
              "MozTransform": "rotateZ(90deg)",
              "MozTransformOrigin": "45% 50%",
              "MozTransition": "150ms",
              "WebkitTransform": "rotateZ(90deg)",
              "WebkitTransformOrigin": "45% 50%",
              "WebkitTransition": "150ms",
              "color": "#268bd2",
              "fontSize": "0.75em",
              "lineHeight": "1.1em",
              "marginLeft": 0,
              "position": "relative",
              "transform": "rotateZ(90deg)",
              "transformOrigin": "45% 50%",
              "transition": "150ms",
            }
          }
        >
          â–¶
        </div>
      </div>
      <label
        onClick={[Function]}
        style={
          Object {
            "MozUserSelect": "inherit",
            "WebkitUserSelect": "inherit",
            "color": "#268bd2",
            "cursor": "pointer",
            "display": "inline-block",
            "margin": 0,
            "padding": 0,
          }
        }
      >
        <span>
          root
          :
        </span>
      </label>
      <span
        onClick={[Function]}
        style={
          Object {
            "color": "#657b83",
            "cursor": "default",
            "paddingLeft": "0.5em",
          }
        }
      >
        <span>
          <span
            style={
              Object {
                "marginLeft": "0.3em",
                "marginRight": "0.3em",
              }
            }
          >
            {}
          </span>
           
          2 keys
        </span>
      </span>
      <ul
        style={
          Object {
            "display": "block",
            "listStyle": "none",
            "margin": 0,
            "padding": 0,
          }
        }
      >
        <li
          style={
            Object {
              "MozUserSelect": "text",
              "WebkitUserSelect": "text",
              "marginLeft": "0.875em",
              "paddingLeft": "2.125em",
              "paddingRight": 0,
              "paddingTop": "0.25em",
              "textIndent": "-0.5em",
              "wordBreak": "break-all",
              "wordWrap": "break-word",
            }
          }
        >
          <label
            style={
              Object {
                "color": "#268bd2",
                "display": "inline-block",
                "margin": "0 0.5em 0 0",
              }
            }
          >
            <span>
              name
              :
            </span>
          </label>
          <span
            style={
              Object {
                "color": "#859900",
              }
            }
          >
            "Juan Dan"
          </span>
        </li>
        <li
          style={
            Object {
              "MozUserSelect": "text",
              "WebkitUserSelect": "text",
              "marginLeft": "0.875em",
              "paddingLeft": "2.125em",
              "paddingRight": 0,
              "paddingTop": "0.25em",
              "textIndent": "-0.5em",
              "wordBreak": "break-all",
              "wordWrap": "break-word",
            }
          }
        >
          <label
            style={
              Object {
                "color": "#268bd2",
                "display": "inline-block",
                "margin": "0 0.5em 0 0",
              }
            }
          >
            <span>
              email
              :
            </span>
          </label>
          <span
            style={
              Object {
                "color": "#859900",
              }
            }
          >
            "jgarcialopez837@gmail.com"
          </span>
        </li>
      </ul>
    </li>
  </ul>,
]
`;
